### PR TITLE
Terraform provider for Kilt transforms

### PR DESCRIPTION
**Pull request by gnosek**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area tests

> /area proposals

**What this PR does / why we need it**:

This PR adds support for modifying terraform Fargate task definitions with kilt.

### CODE REVIEW

File: runtimes/terraform/tf/data_source_fargate_container_definitions.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
+package tf
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/falcosecurity/kilt/runtimes/cloudformation/cfnpatcher"
+	"github.com/hashicorp/terraform-plugin-framework/schema"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
```
Comment (admiral0) [inline runtimes/terraform/tf/data_source_fargate_container_definitions.go:10]: use goimports to split imports between library, external and internal imports

File: runtimes/terraform/tf/fargate.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
+package tf
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"github.com/hashicorp/terraform-plugin-framework/types"
```
Comment (admiral0) [inline runtimes/terraform/tf/fargate.go:7]: use goimports

File: runtimes/terraform/tf/fargate.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+)
+import "github.com/falcosecurity/kilt/runtimes/cloudformation/cfnpatcher"
```
Comment (admiral0) [inline runtimes/terraform/tf/fargate.go:9]: use only 1 import directive

File: runtimes/terraform/tf/fargate.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+type cfProperties struct {
```
Comment (admiral0) [inline runtimes/terraform/tf/fargate.go:11]: I would use `cfnProperties`

File: runtimes/terraform/tf/fargate.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+	RequiresCompatibilities []string                 `json:"RequiresCompatibilities"`
+	ContainerDefinitions    []map[string]interface{} `json:"ContainerDefinitions"`
+}
+
+type cfResource struct {
```
Comment (admiral0) [inline runtimes/terraform/tf/fargate.go:16]: use `cfn` prefix please

File: runtimes/terraform/tf/fargate.go [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+	ResourceType string       `json:"Type"`
+	Properties   cfProperties `json:"Properties"`
+}
+
+type cfStack struct {
+	Resources map[string]cfResource `json:"Resources"`
+}
+
+func patchFargateTaskDefinition(ctx context.Context, containerDefinitions *types.String, kiltConfig *cfnpatcher.Configuration) (patched types.String, err error) {
```
Comment (admiral0) [inline runtimes/terraform/tf/fargate.go:25]: I don't like having named outputs. See [this recommendation](https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_avoid_named_return_values).

File: runtimes/terraform/tf/provider.go [hunk 7]
```diff
@@ -0,0 +1,50 @@
+package tf
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-framework/schema"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
```
Comment (admiral0) [inline runtimes/terraform/tf/provider.go:8]: use `goimports` please

**Review by admiral0 — CHANGES_REQUESTED** [review]

File: runtimes/terraform/tf/fargate.go (hunk 6 above)
Comment (gnosek) [reply runtimes/terraform/tf/fargate.go:25]: I don't like it either. I couldn't find any different way to return a value when cfnpatcher.Patch panics.

**Review by admiral0 — APPROVED** [review]

**Pull request merged**